src/scales/scale.time.js

```diff
@@ -315,15 +333,26 @@ module.exports = function(Chart) {
 
 			var maxTicks = me.getLabelCapacity(minTimestamp || dataMin);
 			var unit = timeOpts.unit || determineUnit(timeOpts.minUnit, minTimestamp || dataMin, maxTimestamp || dataMax, maxTicks);
+			var units = Object.keys(interval);
+			var majorUnit = null;
+			var unitIndex = units.indexOf(unit);
+			if (unitIndex < units.length - 1) {
+				majorUnit = units[unitIndex + 1];
+			}
+
 			me.displayFormat = timeOpts.displayFormats[unit];
+			me.seniorDisplayFormat = timeOpts.displayFormats[majorUnit];
 
 			var stepSize = timeOpts.stepSize || determineStepSize(minTimestamp || dataMin, maxTimestamp || dataMax, unit, maxTicks);
+
 			me.ticks = Chart.Ticks.generators.time({
 				maxTicks: maxTicks,
 				min: minTimestamp,
 				max: maxTimestamp,
 				stepSize: stepSize,
+				majorUnit: majorUnit,
 				unit: unit,
+				timeOpts: timeOpts,
```

Comment:
it's weird that here we pass in `timeOpts` and on the next line we pass in `timeOpts.isoWeekday`. we should maybe change this to `timeOpts.round` for consistency?